packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -105,6 +107,10 @@ export class MonacoEditorModel implements IResolvedTextEditorModel, TextEditorDo
         this.resolveModel = this.readContents().then(
             content => this.initialize(content || '')
         );
+        this.toDispose.push(this.connectionStatusService.onStatusChange(() => {
+            this.cancelSave();
+            this.cancelSync();
+        }));
```

Comment:
The this is where the problem manifested itself, but I wonder if the the problem shouldn't be fixed in the `FileResource` code, since that's where the saving happens, and it's the connection to the backend file system that breaks saving / loading new content?